app/routes.py

```diff
@@ -46,3 +45,13 @@ def search():
     return render_template(
         "recipe.html", recipes=blog_recipes, selected_recipe=new_recipe
     )
+
+
+@app.route("/search-by-dish")
+def search_by_dish():
+    return search("dish")
+
+
+@app.route("/search-by-ingredient")
+def search_by_ingredient():
+    return search("ingredient")
```

Comment:
Nice use of a shared search function for the two endpoints. Changing the route to /search-by-dish breaks the `test_search_route` unit test. The unit test will need to be changed to properly test the /search-by-dish route.